tests/test_helpers.py

```diff
@@ -81,3 +96,171 @@ def test_get_font_stack_add_emoji_true(font_stack_names):
     assert all(
         extended_emoji.issubset(_get_font_stack(name, add_emoji=True)) for name in font_stack_names
     )
+
+
+@pytest.mark.parametrize(
+    "units, x_out",
+    [
+        ("x10^7 m s^-1", ["x10^7", "m", "s^-1"]),
+        (
+            "-20 kg^2 m_12 m[_0^2] g/L %C6H12O6% x10^-3",
+            ["-20", "kg^2", "m_12", "m[_0^2]", "g/L", "%C6H12O6%", "x10^-3"],
+        ),
+        (
+            "  -20  _kg_^2      *m*_12  m[_0^2] g/L%C6H12O6% x10^-3",
+            ["-20", "_kg_^2", "*m*_12", "m[_0^2]", "g/L%C6H12O6%", "x10^-3"],
+        ),
+        (
+            "m^-3 :cdot: kg :cdot: m[_0^2] *per:space:mille*",
+            ["m^-3", ":cdot:", "kg", ":cdot:", "m[_0^2]", "*per:space:mille*"],
+        ),
+        ("", []),
+        (" ", []),
+    ],
+)
+def assert_generate_tokens_list(units: str, x_out: str):
+
+    x = _generate_tokens_list(units_notation=units)
+    assert x == x_out
+
+
+@pytest.mark.parametrize(
+    "content, x_out",
+    [
+        ("2", '<span style="white-space:nowrap;"><sub>2</sub></span>'),
+        ("2*e*5", '<span style="white-space:nowrap;"><sub>2*e*5</sub></span>'),
+        ("", '<span style="white-space:nowrap;"><sub></sub></span>'),
+    ],
+)
+def assert_units_to_subscript(content: str, x_out: str):
+    x = _units_to_subscript(content=content)
```

Comment:
`_units_to_subscript()` is a single line function that produces a string. I don't understand what new thing is being tested by e.g. using both `"2"` and `"2*e*5"` in the tests.

Reply:
I removed the latter since it was redundant.